### ISSUE TITLE
Searching uses values hidden by numericFormat

### ISSUE THREAD

**Issue by JustinCreasy** [handsontable/handsontable#4884]
### Description
If this is the intended functionality this issue can be closed.

When using the search feature against a column that is type: "numeric" and has a numericFormat pattern: '0.0000' but values that go beyond 4 decimal places, the search will still highlight cells with matching source values that are not displayed.

### Steps to reproduce

1. Go to: https://jsfiddle.net/jcreasy/xpjgLr9n/10/
2. Enter '9' into the search field at the top
3. Notice that the displayed value 0.1111 gets highlighted, it's source value is .1111199

### Demo

https://jsfiddle.net/jcreasy/xpjgLr9n/10/

### Your environment
* Handsontable version: Pro 1.16/1.17 (referencing static version and you updated today)
* Browser Name and version: Chrome, Firefox, Edge
* Operating System: Windows 10

**Comment by AMBudnik** [comment]
I do not think that this is an intended behavior. Surely we get the source data but if the 

```
 numericFormat: {
     pattern: '0.0000'
 }
```

the result should not highlight.

I am adding a `bug` and `to be discussed` labels for other team members to share their thoughts.

Thank you for sharing the report and demo @JustinCreasy

**Comment by pnowak** [comment]
`search` plugin operates in the `query` method on `getDataAtCell`, which returns the cell value. 
This means that it takes the whole value, not the one visible after formatting.
But we could think about change this default work. It needs further investigation.

**Comment by conschtek** [comment]
Hi,
is there any update on this topic? The issue also occurres when custom renderers are used, which changes the appearance of the data. I'm currently using v12.3.1 and I'm interested if it maybe got fixed in a newer version.
Thank you

**Comment by adrianszymanski89** [comment]
Hi @conschtek 

This issue isn't on our roadmap yet. We will update this thread once it's done.

**Comment by sl01k** [comment]
The search plugin queries against raw source data by design. To match against the displayed (formatted) value, you can use a custom queryMethod:

```
  const matchAgainstRendered = (queryStr, value, cellProperties) => {
    if (!queryStr || value === null || value === undefined) return false;

    const query = String(queryStr).trim().toLowerCase();
    let renderedValue;

    if (
      cellProperties?.type === 'numeric' &&
      cellProperties?.numericFormat?.pattern
    ) {
      renderedValue =
  numbro(value).format(cellProperties.numericFormat.pattern);
    } else {
      renderedValue = String(value);
    }

    return String(renderedValue).toLowerCase().includes(query);
  };

  const hot = new Handsontable(element, {
    // ...
    search: {
      queryMethod: matchAgainstRendered
    }
  });
```
Demo: https://jsfiddle.net/9gmy07j5/7/

This approach also works for custom renderers — just apply the same formatting logic inside queryMethod.

Closing as resolved via custom queryMethod.
